src/fuzz/fuzz-journal-remote.c

```diff
@@ -46,8 +47,13 @@ int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
         r = sd_journal_open_files(&j, (const char**) STRV_MAKE(name), 0);
         assert_se(r >= 0);
 
-        r = show_journal(stdout, j, OUTPUT_VERBOSE, 0, 0, -1, 0, NULL);
-        assert_se(r >= 0);
+        for (mode = 0; mode < _OUTPUT_MODE_MAX; mode++) {
+                r = show_journal(stdout, j, mode, 0, 0, -1, 0, NULL);
+                assert_se(r >= 0);
+
+                r = sd_journal_seek_head(j);
+                assert_se(r >= 0);
```

Comment:
In http://llvm.org/docs/LibFuzzer.html#fuzz-target It is recommended that logging should be avoided. I'm wondering if it's possible to make `show_journal` skip calling `fprintf` if it's run in the "fuzzer" mode.

Reply:
Half of this fuzzer is the journal output code, and writing the stuff out is an integral part of it. Disabling output would be possible, but it would reduce the coverage a lot.

Reply:
In the meantime, I reconsidered, and pushed another patch on top to write the logs to /dev/null. I *think* all our output code ignored write failures, so this should have the same effect and writing to stdout. I tested that the one failure that was found is still reproducible with /dev/null.

Unfortunately there seems to be some leak of fds. I need to work on this a bit more.